### ISSUE TITLE
Recursive calls on Builder.from when extending object

### ISSUE THREAD

**Issue by cocoviper** [immutables/immutables#128]
Here i have my 2 objects. RoomBasic is a simple abstract class which is used as basis for other Immutable classes such as AbstractRoomServer.
```java
abstract class RoomBasic implements Serializable {
    abstract String getName();
    abstract boolean isPasswordRoom();
}

abstract class AbstractRoomServer extends RoomBasic {
    abstract CopyOnWriteArrayList<AbstractClientBasicInfo> getClientBasicInfos();
    abstract String getPassword();
}
```
When AbstractRoomServer's immutable is generated I get the following build().from() methods. The private method from(Object object) is never called instead the 2 other from methods call themselves recursively.
```java
    public final Builder from(RoomBasic instance) {
      Objects.requireNonNull(instance);
      from(instance);
      return this;
    }
 
    public final Builder from(AbstractRoomServer instance) {
      Objects.requireNonNull(instance);
      from(instance);
      return this;
    }
  
    private void from(Object object) {
      if (object instanceof RoomBasic) {
        RoomBasic instance = (RoomBasic) object;
        setName(instance.getName());
      }
      if (object instanceof AbstractRoomServer) {
        AbstractRoomServer instance = (AbstractRoomServer) object;
        setClientBasicInfos(instance.getClientBasicInfos());
        setPassword(instance.getPassword());
        setHash(instance.getHash());
      }
    }
```

Thank you for your time!

**Comment by augustotravillio** [comment]
Thank you for reporting this, definitely a bug. @elucash could you check the commit and maybe release the fix?

**Comment by elucash** [comment]
easy fix! I wonder if there are other problems, at least I see set of attributes in definition and in builder from method do not match in provided snippet. Is it another bug?
```java
    private void from(Object object) {
      if (object instanceof RoomBasic) {
        RoomBasic instance = (RoomBasic) object;
        setName(instance.getName());
        // setPasswordRoom(instance.isPasswordRoom()); <-- this attribute is in definition, but missing here
      }
      if (object instanceof AbstractRoomServer) {
        AbstractRoomServer instance = (AbstractRoomServer) object;
        setClientBasicInfos(instance.getClientBasicInfos());
        setPassword(instance.getPassword());
        setHash(instance.getHash()); // <-- don't see getHash() accessor in definition
      }
    }
```

**Comment by cocoviper** [comment]
The definition isPasswordRoom and getHash are defined in AbstractRoomServer as followed sorry for the confusion.

```java
@Value.Immutable
abstract class AbstractRoomServer extends RoomBasic {
    abstract CopyOnWriteArrayList<AbstractClientBasicInfo> getClientBasicInfos();
    abstract String getPassword();
    
    @Value.Default
    public String getHash(){
        return UUID.randomUUID().toString();
    }

    @Override boolean isPasswordRoom() {
        return !getPassword().isEmpty();
    }

}

```

**Comment by elucash** [comment]
Ok, we will then just release the fix later today. Thanks!

**Comment by elucash** [comment]
Released in 2.0.16
